apidef/oas/operation.go

```diff
@@ -163,6 +166,130 @@ func (s *OAS) fillPathsAndOperations(ep apidef.ExtendedPathsSet) {
 	s.fillDoNotTrackEndpoint(ep.DoNotTrackEndpoints)
 	s.fillRequestSizeLimit(ep.SizeLimit)
 	s.fillRateLimitEndpoints(ep.RateLimit)
+	s.fillMockResponsePaths(s.Paths, ep)
+}
+
+// fillMockResponsePaths converts classic API mock responses to OAS format.
+// This method only handles direct mock response conversions, as other middleware
+// configurations (like allow lists, block lists, etc.) are converted to classic
+// API mock responses in an earlier step of the process.
+//
+// For each mock response, it:
+// 1. Creates an OAS operation with a unique ID (if it doesn't exist)
+// 2. Sets up the mock response with content type detection and example values
+// 3. Configures the operation to ignore authentication for this endpoint
+//
+// The content type is determined by:
+// - Checking the Content-Type header if present
+// - Attempting to parse the body as JSON
+// - Defaulting to text/plain if neither above applies
+func (s *OAS) fillMockResponsePaths(paths openapi3.Paths, ep apidef.ExtendedPathsSet) {
+	for _, mock := range ep.MockResponse {
+		operationID := s.getOperationID(mock.Path, mock.Method)
+
+		var operation *openapi3.Operation
+
+		for _, item := range paths {
+			if op := item.GetOperation(mock.Method); op != nil && op.OperationID == operationID {
+				operation = op
+				break
+			}
+		}
+
+		if operation.Responses == nil {
+			operation.Responses = openapi3.NewResponses()
+		}
+
+		// Response description is required by the OAS spec, but we don't have it in Tyk classic.
+		// So we're using a dummy value to satisfy the spec.
+		dummyDescription := "CHANGE ME"
+
+		response := &openapi3.Response{
+			Headers:     make(openapi3.Headers),
+			Description: &dummyDescription,
+		}
+
+		contentType := "text/plain"
+
+		// We attempt to guess the content type by checking if the body is a valid JSON.
+		if mock.Body != "" {
+			var jsonValue = []interface{}{
+				map[string]json.RawMessage{},
+				[]json.RawMessage{},
+			}
+
+			for _, value := range jsonValue {
+				if err := json.Unmarshal([]byte(mock.Body), value); err == nil {
+					contentType = "application/json"
+					break
+				}
+			}
+		}
+
+		// Check Content-Type header if present (takes precedence over auto-detection)
+		for name, value := range mock.Headers {
+			if http.CanonicalHeaderKey(name) == tykheader.ContentType {
+				contentType = value
+				break
```

Comment:
if you split the method into multiple helpers. you will be able to just return from the underlying helper instead of breaking the for loop

Reply:
Fixed, thank you. PTAL again